backend/test/integration/test_waiting_list.py

```diff
@@ -49,6 +48,34 @@ def test_already_in_list(self, with_db, with_client, make_waiting_list):
             # Ensure we did not send out an email
             mock.assert_not_called()
 
+    def test_rate_limited(self, with_db, with_client):
+        """Make sure our rate limit kicks in after 2 requests."""
+
+        # Oops, they messed up the first sign-up
+        email = 'incorrect_spelling_oops@example.org'
+        response = with_client.post('/waiting-list/join', json={'email': email})
+
+        # Ensure the response was okay!
+        assert response.status_code == 200, response.json()
+        assert response.json() is True
+
+        # Quickly recover with the right email
+        email = 'hello@example.org'
+        response = with_client.post('/waiting-list/join', json={'email': email})
+
+        # Ensure the response was okay!
+        assert response.status_code == 200, response.json()
+        assert response.json() is True
+
+        # Oh so sneakily signing up a third!
+        email = 'hello2@example.org'
+        response = with_client.post('/waiting-list/join', json={'email': email})
+
+        # Ensure the response was rate limited
+        assert response.status_code == 429, response.json()
+        data = response.json()
+        assert data['detail']['id'] == APIRateLimitExceeded.id_code
```

Comment:
Why does the `@limiter.limit("2/minute")` for this not work here? It returns 200 in tests instead of expected 429?

Reply:
that's not right. I'll look into this next week.